build.gradle

```diff
@@ -27,3 +36,49 @@ task sourcesJar(type: Jar, dependsOn: classes) {
 	classifier = 'sources'
 	from sourceSets.main.allSource
 }
+
+// create a javadoc file for publishing
+task javadocJar(type: Jar, dependsOn: javadoc) {
```

Comment:
Do we really need a javadoc?

Reply:
well I thought it'd be a nice-to-have, mainly for including it in the maven publication

Reply:
It adds quite a bit more to the build.gradle (due to the fix bellow). And there are source jars anyway so when you depend on a mod it should give you sources.